var/spack/repos/builtin/packages/nmap/package.py

```diff
@@ -32,14 +33,20 @@ class Nmap(AutotoolsPackage):
 
     variant("liblua", default=True, description="Enable lua (required by all of NSE)")
     variant("ncat", default=True, description="Enable ncat")
-    variant("ndiff", default=True, description="Enable ndiff")
+    variant("ndiff", default=False, description="Enable ndiff")
     variant("nping", default=True, description="Enable nping")
-    variant("zenmap", default=True, description="Enable zenmap")
+    variant("zenmap", default=False, description="Enable zenmap")
     variant("nmap-update", default=False, description="Enable nmap-update")
 
     depends_on("openssl@1.1:", when="@7.50:")
     depends_on("openssl@:1.0.9", when="@:7.49")
 
+    # python2 is not packaged with spack so these features are disabled unless a
+    # user has an external python2 package configured.
+    # this will be fixed in the next release of nmap
+    depends_on("python@:2", when="+ndiff")
```

Comment:
Hi David! Thanks so much for this contribution!!!

(1) What version of nmap is expected to fix this? It would be preferable to add a version constraint with an upper bound to the `when` clause in order to shut off this dependency for later nmap versions automatically. Please let me know if the documentation on version ranges is useful as I am working on it in a separate PR.

(2) What link did you use to determine this information (so we can see if the expected release date changes)? I use nmap infrequently but might file as maintainer.

Reply:
Funny seeing you here haha!

(1) The next release of nmap will add py3 support to their internal tools. I can add a version constraint for this dep easily here.

(2) I determined this from digging around in their repo, and this changelog: https://nmap.org/changelog.html
(2.1) Their configure script explicitly checks for python2 up until the unreleased version

To reply to both comments (@tldahlgren) do you all prefer disabling these features completely until the next nmap release, or should we keep the py2 dep but gate it by version? Either way changes will need to be made at update time so I sort of prefer just removing these feature flags until the next release

Reply:
Great question.  I'm torn as I don't like completely removing an option that can soon be restored.  But purists won't want the code until it/they are available.

Reply:
I pushed the "just remove the features" version, but I'm seeing a different build issue now on my linux machine, and on my m1 mac.

On linux there's an openssl issue, on m1 there seems to be a build system race. I'll keep poking at this